src/clients/c++/perf_client/perf_client.cc

```diff
@@ -259,142 +275,293 @@ Usage(char** argv, const std::string& msg = std::string())
+                   "sample a time interval specified by -p and take "
+                   "measurement over the requests completed within that time "
+                   "interval. The default value is 5000 msec.",
+                   18)
             << std::endl;
-  std::cerr << "The --max-threads flag sets the maximum number of threads that"
-            << " will be created for providing desired concurrency."
-            << " Default is 16." << std::endl;
   std::cerr
-      << "For -t, it indicates the number of starting concurrent requests if -d"
-      << " flag is set." << std::endl;
+      << FormatMessage(
+             " --search-range <m:n:o>: Determines the range of concurrency "
+             "levels covered by the perf_client. The perf_client will start "
+             "from the concurrency level of 'm' to 'n' with steps of 'o'. The "
+             "default value of 'm' and 'o' are 1. If 'n' is not specified then "
+             "perf_client will run for a single concurrency level determined "
+             "by 'm'. If 'n' is set as 0, then the concurrency limit will be "
+             "incremented by 'o' till latency threshold is met. 'n' and -l can "
+             "not be both 0 simultaneously.",
+             18)
+      << std::endl;
+  std::cerr << FormatMessage(
+                   " --latency-threshold (-l): Sets the limit on the observed "
+                   "latency. Client will top incrementing the concurrency and "
+                   "abort the execution once the measured latency exceeds this "
+                   "threshold. By default, latency threshold is set 0 and "
+                   "the perf_client will run for entire --search-range.",
+                   18)
+            << std::endl;
   std::cerr
-      << "For -s, it indicates the deviation threshold for the measurements. "
-         "The measurement is considered as stable if the recent 3 measurements "
-         "are within +/- (deviation threshold)% of their average in terms of "
-         "both infer per second and latency. Default is 10(%)"
+      << FormatMessage(
+             " --max-threads: Sets the maximum number of threads that will be "
+             "created for providing desired concurrency. Default is 16.",
+             18)
       << std::endl;
   std::cerr
-      << "For -c, it indicates the maximum number of concurrent requests "
-         "allowed if -d flag is set. Once the number of concurrent requests "
-         "exceeds the maximum, the perf client will stop and exit regardless "
-         "of the latency threshold. Default is 0 to indicate that no limit is "
-         "set on the number of concurrent requests."
+      << FormatMessage(
+             " --stability-offset (-s): It indicates the deviation threshold "
+             "for the measurements. The measurement is considered as stable if "
+             "the recent 3 measurements are within +/- (deviation threshold)% "
```

Comment:
Don't say "deviation threshold" when the flag is called "stability percentage", use "stability percentage" in the description.

Reply:
You still say "deviation threshold" which doesn't have any standard meaning. "Indicates the allowed variation in latency measurements when determining if a result is stable. The measurement is ... within +/- (stability percentage) ..."